Feature/4355 channel value rnpv5

surajeaton opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
Fixes # BLUI-4355

<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Fixed Channel Value icon issue
- Updated channel value text variants
- Added test case

<!-- Include screenshots if they will help illustrate the changes in this PR -->
#### Screenshots / Screen Recording (if applicable)
- 
![Simulator Screenshot - iPhone 14 - 2023-09-13 at 16 12 21](https://github.com/etn-ccis/blui-react-native-component-library/assets/120575281/aae72df6-323c-4132-b5e3-c47708bf8518)

<!-- Instruction for PR reviewers, if more complicated than a simple yarn start -->
#### To Test:
- cd MD3
- yarn ios

<!-- Useful for draft pull requests -->
#### Any specific feedback you are looking for?
-

daileytj commented on components/src/md3/ChannelValue/ChannelValue.tsx:
```diff
@@ -2,7 +2,7 @@ import React, { useCallback } from 'react';
 import { MD3Theme, Text, useTheme } from 'react-native-paper';
 import { $DeepPartial } from '@callstack/react-theme-provider';
-import { Icon } from '../Icon';
+import { Icon } from '../icon';
```
I just noticed the build is outputting the icon folder with a lowercase. Not sure why this is happening, but if this change works then we can roll with it.

daileytj commented on components/src/md3/ChannelValue/ChannelValue.tsx:
```diff
@@ -110,7 +110,7 @@ export const ChannelValue: React.FC<ChannelValueProps> = (props) => {
         if (icon) {
             return (
-                <View style={[{ marginRight: Math.round(fontSize / 3) }]}>
+                <View style={[{ marginRight: Math.round(fontSize / 3), paddingTop: 2 }]}>
```
I don't think this is the appropriate fix for the icon alignment with the text. If you look at the showcase vs the MD3 demo you'll see that the showcase examples are now off. Not sure why we're getting different results, but showcase is where we should be testing things now and the icon is not aligning properly now.

daileytj replied:
The text seems to be aligned without any additional styles in the showcase

daileytj commented:
I added a branch `feature/remove-unnecessary-md3-demo` to remove the demo and cleanup the icon file names and remove the ChannelValue padding. Feel free to look that over and possibly merge that into this branch if everything builds appropriately and you are able to work in the showcase. I added a showcase branch `feature/update-channelvalue-demo` as well to address the Channelvalue icon image example. 

We can address all these changes separately as well, but I think it would be good to get rid of that additional MD3 demo folder sooner than later.

surajeaton commented:
> I added a branch `feature/remove-unnecessary-md3-demo` to remove the demo and cleanup the icon file names and remove the ChannelValue padding. Feel free to look that over and possibly merge that into this branch if everything builds appropriately and you are able to work in the showcase. I added a showcase branch `feature/update-channelvalue-demo` as well to address the Channelvalue icon image example.
> 
> We can address all these changes separately as well, but I think it would be good to get rid of that additional MD3 demo folder sooner than later.

Done. Please review.

huayunh commented:
- Can you update the channel value to use a "semibold" font weight by default? Sorry I made some bad design decisions in the component sticker sheet.
- Are the CI test cases expected to fail? Or am I scope-creeping?
- To avoid legal troubles, ensure that the images you are using are copyright free. Talking about MD3/assets/images/test-avatar.png

surajeaton commented:
> * Can you update the channel value to use a "semibold" font weight by default? Sorry I made some bad design decisions in the component sticker sheet.
> * Are the CI test cases expected to fail? Or am I scope-creeping?
> * To avoid legal troubles, ensure that the images you are using are copyright free. Talking about MD3/assets/images/test-avatar.png

- Updated the font weight in channel value.
- We might need to fix circle ci in separate pr, as it is failing due to various reasons, sometimes Install Chromedriver sometimes snapshots (even if the snapshots are updated using `yarn test -u` command)
- MD3/assets/images/test-avatar.png is taken from showcase of same repo

huayunh commented:
> Updated the font weight in channel value

I saw that there's this code snippet 
```
                        {
                            color: getColor(),
                            fontSize: fontSize,
                            letterSpacing: 0,
                            fontFamily: 'OpenSans-Regular',
                            fontWeight: '600',
                        },
```
* Why is `letterSpacing` enforced here? 
* I see that we have both `fontFamily: 'OpenSans-Regular'` and `fontWeight: '600'` instead of just one `fontFamily: 'OpenSans-Semibold'` (or whatever the font is called). Is that a technical constraint?

> MD3/assets/images/test-avatar.png is taken from showcase of same repo

So are you sure that the showcase image was copyright-free to begin with? Perhaps @daileytj knows it better?

daileytj commented:
@huayunh @surajeaton I updated the showcase and MD3 demo to use a public domain image

surajeaton commented:
> * fontFamily: 'OpenSans-Semibold

Removed letterSpacing.
None of the fontFamily are updating the font weight (always 400) when passed as `fontFamily: 'OpenSans-Regular' or fontFamily: 'OpenSans-Semibold'.

We may need to consider it as an action item to fix it across the library.

huayunh commented:
interesting...

I am not sure why MD3/components/ChannelValue.tsx was not getting updated, but the rest seems good to me

daileytj commented:
> interesting...
> 
> I am not sure why MD3/components/ChannelValue.tsx was not getting updated, but the rest seems good to me

Everyone should be working from src/md3, I need to remove the MD3 folder entirely. I will do that after we get the rest of the currently worked-on components in.

daileytj commented:
![Simulator Screenshot - iPhone SE (3rd generation) - 2023-09-18 at 08 50 27](https://github.com/etn-ccis/blui-react-native-component-library/assets/13989985/2264195d-6a4a-4e81-9753-88246b56887d)
![Simulator Screenshot - iPhone SE (3rd generation) - 2023-09-18 at 08 54 05](https://github.com/etn-ccis/blui-react-native-component-library/assets/13989985/58afa254-c477-4c38-96ad-f80f9449ca26)
![Simulator Screenshot - iPhone SE (3rd generation) - 2023-09-18 at 08 55 27](https://github.com/etn-ccis/blui-react-native-component-library/assets/13989985/f06d1fe4-5c21-42f9-81eb-4fa9a462298f)
![Simulator Screenshot - iPhone SE (3rd generation) - 2023-09-18 at 08 59 52](https://github.com/etn-ccis/blui-react-native-component-library/assets/13989985/0c6c5095-8e20-4481-a3b8-4dafdb94f297)

 For some reason the semi bold and the regular fonts look the same despite looking as though they are properly configured. At this point I say we leave it as is with regular font family and font weight 600 and address this at a later time.
